From 6915f1a190d820642fce956b84ddc909fc1fba26 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:39:11 -0400
Subject: [alsa] hda: init_flag is in struct hda_intel

Message-id: <140972823573-393-patchset-jkysela@redhat.com>
Patchwork-id: 91089
O-Subject: [RHEL 7.1 PATCH 392/442] ALSA: hda - init_flag is in struct hda_intel
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

Since init_failed flag was moved to struct hda_intel, its access in
the commit [4da63c6f: ALSA: hda - Fix broken PM due to incomplete
i915 initialization] is also replaced with hda->init_failed
appropriately.

(picked from kernel: 1618e84aa87af94b46921b265b68bb7994571c20)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/hda_intel.c b/sound/pci/hda/hda_intel.c
index 4288cc4..61aba63 100644
--- a/sound/pci/hda/hda_intel.c
+++ b/sound/pci/hda/hda_intel.c
@@ -773,9 +773,10 @@ static int azx_suspend(struct device *dev)
  struct pci_dev *pci = to_pci_dev(dev);
  struct snd_card *card = dev_get_drvdata(dev);
  struct azx *chip = card->private_data;
+ struct hda_intel *hda = container_of(chip, struct hda_intel, chip);
  struct azx_pcm *p;
 
- if (chip->disabled || chip->init_failed)
+ if (chip->disabled || hda->init_failed)
   return 0;
 
  snd_power_change_state(card, SNDRV_CTL_POWER_D3hot);
@@ -806,8 +807,9 @@ static int azx_resume(struct device *dev)
  struct pci_dev *pci = to_pci_dev(dev);
  struct snd_card *card = dev_get_drvdata(dev);
  struct azx *chip = card->private_data;
+ struct hda_intel *hda = container_of(chip, struct hda_intel, chip);
 
- if (chip->disabled || chip->init_failed)
+ if (chip->disabled || hda->init_failed)
   return 0;
 
  if (chip->driver_caps & AZX_DCAPS_I915_POWERWELL) {
@@ -843,8 +845,9 @@ static int azx_runtime_suspend(struct device *dev)
 {
  struct snd_card *card = dev_get_drvdata(dev);
  struct azx *chip = card->private_data;
+ struct hda_intel *hda = container_of(chip, struct hda_intel, chip);
 
- if (chip->disabled || chip->init_failed)
+ if (chip->disabled || hda->init_failed)
   return 0;
 
  if (!(chip->driver_caps & AZX_DCAPS_PM_RUNTIME))
@@ -867,11 +870,12 @@ static int azx_runtime_resume(struct device *dev)
 {
  struct snd_card *card = dev_get_drvdata(dev);
  struct azx *chip = card->private_data;
+ struct hda_intel *hda = container_of(chip, struct hda_intel, chip);
  struct hda_bus *bus;
  struct hda_codec *codec;
  int status;
 
- if (chip->disabled || chip->init_failed)
+ if (chip->disabled || hda->init_failed)
   return 0;
 
  if (!(chip->driver_caps & AZX_DCAPS_PM_RUNTIME))
@@ -907,8 +911,9 @@ static int azx_runtime_idle(struct device *dev)
 {
  struct snd_card *card = dev_get_drvdata(dev);
  struct azx *chip = card->private_data;
+ struct hda_intel *hda = container_of(chip, struct hda_intel, chip);
 
- if (chip->disabled || chip->init_failed)
+ if (chip->disabled || hda->init_failed)
   return 0;
 
  if (!power_save_controller ||
-- 
1.7.1